test/matrix.cc

```diff
@@ -286,30 +355,28 @@ TEST(Matrix, test_timedistancematrix_results_sequence) {
       CreateSimpleCost(request.options().costings().find(request.options().costing_type())->second);
 
   TimeDistanceMatrix timedist_matrix;
-  std::vector<TimeDistance> results =
-      timedist_matrix.SourceToTarget(*request.mutable_options()->mutable_sources(),
-                                     *request.mutable_options()->mutable_targets(), reader,
-                                     mode_costing, sif::TravelMode::kDrive, 400000.0);
-
-  // expected results are the same as `matrix_answers`, but without the last column
-  std::vector<TimeDistance> expected_results = {
-      {28, 28},     {2027, 1837}, {2403, 2213}, {1519, 1398}, {1808, 1638}, {2061, 1951},
-      {2311, 2111}, {701, 641},   {0, 0},       {5562, 5177}, {3952, 3707}, {4367, 4107},
-  };
-
-  for (uint32_t i = 0; i < results.size(); ++i) {
-    EXPECT_NEAR(results[i].dist, expected_results[i].dist, kThreshold)
-        << "result " + std::to_string(i) + "'s distance is not equal to" +
-               " the expected value for TimeDistMatrix";
-
-    EXPECT_NEAR(results[i].time, expected_results[i].time, kThreshold)
-        << "result " + std::to_string(i) +
-               "'s time is not equal to the expected value for TimeDistMatrix";
+  timedist_matrix.SourceToTarget(request, reader, mode_costing, sif::TravelMode::kDrive, 400000.0);
+  auto& matrix = request.matrix();
+
+  // expected results are the same as `matrix_answers`, but without the last target
+  // clang-format off
+  std::vector<std::vector<uint32_t>> expected_results = {{28, 28},     {2027, 1837}, {2403, 2213}, 
+                                                      {1519, 1398}, {1808, 1638}, {2061, 1951},
+                                                      {2311, 2111}, {701, 641},   {0, 0},
+                                                      {5562, 5177}, {3952, 3707}, {4367, 4107}};
+  // clang-format on
+
+  for (uint32_t i = 0; i < matrix.times().size(); ++i) {
+    EXPECT_NEAR(matrix.distances()[i], expected_results[i][1], kThreshold)
+        << "result " + std::to_string(i) + "'s distance is not equal" +
+               " to expected value for TDMatrix";
+
+    EXPECT_NEAR(matrix.times()[i], expected_results[i][0], kThreshold)
+        << "result " + std::to_string(i) + "'s time is not equal" + " to expected value for TDMatrix";
   }
 }
 
-// TODO: it was commented before. Why?
-TEST(Matrix, DISABLED_test_matrix_osrm) {
+TEST(Matrix, test_matrix_osrm) {
```

Comment:
to increase coverage, I test the OSRM matrix format properly. I guess the reason why it commented was because it wasn't set up to actually test OSRM format, it just looked at the raw time/distances, not at the json output